zkevm-circuits/src/root_circuit/aggregation.rs

```diff
@@ -0,0 +1,713 @@
+            let pi = meta.instance_column();
+            [w_l, w_r, w_o].map(|column| meta.enable_equality(column));
+            meta.create_gate(
+                "q_l·w_l + q_r·w_r + q_o·w_o + q_m·w_l·w_r + q_c + pi = 0",
+                |meta| {
+                    let [w_l, w_r, w_o] =
+                        [w_l, w_r, w_o].map(|column| meta.query_advice(column, Rotation::cur()));
+                    let [q_l, q_r, q_o, q_m, q_c] = [q_l, q_r, q_o, q_m, q_c]
+                        .map(|column| meta.query_fixed(column, Rotation::cur()));
+                    let pi = meta.query_instance(pi, Rotation::cur());
+                    Some(
+                        q_l * w_l.clone()
+                            + q_r * w_r.clone()
+                            + q_o * w_o
+                            + q_m * w_l * w_r
+                            + q_c
+                            + pi,
+                    )
+                },
+            );
+            StandardPlonkConfig {
+                selectors: [q_l, q_r, q_o, q_m, q_c],
+                wires: [w_l, w_r, w_o],
+            }
+        }
+    }
+
+    /// Standard plonk with few assignments for testing purpose.
+    #[derive(Clone, Copy)]
+    pub struct StandardPlonk<F>(F);
+
+    impl<F: FieldExt> StandardPlonk<F> {
+        /// Create a `StandardPlonk` with random instnace.
+        pub fn rand<R: RngCore>(mut rng: R) -> Self {
+            Self(F::from(rng.next_u32() as u64))
+        }
+
+        /// Returns instances.
+        pub fn instances(&self) -> Vec<Vec<F>> {
+            vec![vec![self.0]]
+        }
+    }
+
+    impl<F: FieldExt> Circuit<F> for StandardPlonk<F> {
+        type Config = StandardPlonkConfig;
+        type FloorPlanner = V1;
+
+        fn without_witnesses(&self) -> Self {
+            *self
+        }
+
+        fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+            meta.set_minimum_degree(4);
+            StandardPlonkConfig::configure(meta)
+        }
+
+        fn synthesize(
+            &self,
+            config: Self::Config,
+            mut layouter: impl Layouter<F>,
+        ) -> Result<(), Error> {
+            let [q_l, q_r, q_o, q_m, q_c] = config.selectors;
+            let [w_l, w_r, w_o] = config.wires;
+            layouter.assign_region(
+                || "",
+                |mut region| {
+                    let a = region.assign_advice(|| "", w_l, 0, || Value::known(self.0))?;
+                    region.assign_fixed(|| "", q_l, 0, || Value::known(-F::one()))?;
+                    a.copy_advice(|| "", &mut region, w_r, 1)?;
+                    a.copy_advice(|| "", &mut region, w_o, 2)?;
+                    region.assign_advice(|| "", w_l, 3, || Value::known(-F::from(5)))?;
+                    for (column, idx) in [q_l, q_r, q_o, q_m, q_c].iter().zip(1..) {
+                        region.assign_fixed(|| "", *column, 3, || Value::known(F::from(idx)))?;
+                    }
```

Comment:
It's because currently there is some problem for `halo2wrong` if there is identity point in preprocessed or proof (kind of forget to investigate for a long time), so here it's trying to make sure the fixed columns are not identity by assigning some random values.

I have added some comments and error handling in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1000/commits/54935c5028d8a0245d290cf42ec46974063fe94a, and will try to dig into that issue.